<?xml version="1.0"?>
<!--
  Copyright (C) 2018-2021 LEIDOS.

  Licensed under the Apache License, Version 2.0 (the "License"); you may not
  use this file except in compliance with the License. You may obtain a copy of
  the License at

  http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
  License for the specific language governing permissions and limitations under
  the License.
-->
<launch>  
  <arg name="base_link_frame_id" default="base_link" doc="Frame ID of the base_link frame"/>
  <arg name="map_frame_id" default="map" doc="Frame ID of the map frame"/>
  <arg name="heading_frame_id" default="ned_heading" doc="Frame ID of the frame with +x aligned to vehicle heading and +z into the ground"/>

  <!-- gnss_to_map_convertor Node -->
  <node pkg="gnss_to_map_convertor" type="gnss_to_map_convertor_node" name="gnss_to_map_convertor">
    <param name="base_link_frame_id" value="$(arg base_link_frame_id)"/>
    <param name="map_frame_id" value="$(arg map_frame_id)"/>
    <param name="heading_frame_id" value="$(arg heading_frame_id)"/>
  </node>
</launch>